headerfs/store_test.go

```diff
@@ -579,26 +589,46 @@ func TestFilterHeaderStateAssertion(t *testing.T) {
 		// A header that's well beyond the chain height, it should be a
 		// noop.
 		{
+			name: "assertion not found",
 			headerAssertion: &FilterHeader{
 				Height: 500,
 			},
 			shouldRemove: false,
 		},
 	}
-	for i, testCase := range testCases {
-		fhs, err := NewFilterHeaderStore(
-			tempDir, db, RegularFilter, &chaincfg.SimNetParams,
-			testCase.headerAssertion,
-		)
-		if err != nil {
-			t.Fatalf("unable to make new fhs: %v", err)
-		}
 
-		if testCase.shouldRemove {
-			_, err = os.Stat(fhs.fileName)
-			if err == nil {
-				t.Fatalf("%#v: file should have been removed", i)
+	for _, testCase := range testCases {
+		success := t.Run(testCase.name, func(t *testing.T) {
+			// We'll start the test by setting up our required
+			// dependencies.
+			cleanUp, tempDir, db := setup(t)
+			defer cleanUp()
+
+			// We'll then re-initialize the filter header store with
+			// its expected assertion.
+			fhs, err := NewFilterHeaderStore(
+				tempDir, db, RegularFilter, &chaincfg.SimNetParams,
+				testCase.headerAssertion,
+			)
+			if err != nil {
+				t.Fatalf("unable to make new fhs: %v", err)
+			}
+
+			// If the assertion failed, we should expect the tip of
+			// the chain to no longer exist as the state should've
+			// been removed.
+			_, err = fhs.FetchHeaderByHeight(chainTip)
+			if testCase.shouldRemove {
+				if _, ok := err.(*ErrHeaderNotFound); !ok {
+					t.Fatal("expected file to be removed")
+				}
+			}
+			if !testCase.shouldRemove && err != nil {
```

Comment:
Why are we no longer checking the existence of the file here?

Reply:
Ah nvm, it'll be re-created as a result of the re-initialization of the filter header store.